lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -1872,26 +1870,35 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
           }
         }
 
-        // Handle cases that axes are explicitly specified.
-        // Extract the axes values from the axes operand.
-        // This really shouldn't happen but it helps pass weird tests.
-        // TODO: Derive the chosen axes from the data type and final result type
-        // instead of using the dynamic axes at operand[1].
-        if (!binder.tensorOperandAtIndex(axes, 1)) {
-          Torch::BaseTensorType axesType =
-              axes.getType().cast<Torch::BaseTensorType>();
-          auto sizes = axesType.getSizes();
-          for (int i = 0; i < sizes[0]; i++) {
-            Value selectIndex = rewriter.create<Torch::ConstantIntOp>(
-                binder.getLoc(), rewriter.getI64IntegerAttr(i));
-            Value extract = rewriter.create<Torch::AtenSelectIntOp>(
-                binder.getLoc(),
-                axesType.getWithSizesAndDtype(llvm::SmallVector<int64_t>{1},
-                                              axesType.getOptionalDtype()),
-                axes, zero, selectIndex);
-            Value dim = rewriter.create<Torch::AtenItemOp>(binder.getLoc(),
-                                                           torchIntTy, extract);
-            axesList.push_back(dim);
+        // Derive the chosen axes from the data type and final result type
+        // Fix --convert-torch-to-linalg bug:  `dim` argument must be a
+        // constant int list or None
+        auto resultTypeSizes = resultType.getSizes();
```

Comment:
It is possible there will be more information you can derive by folding on the inputs values. We should not assume the values do not matter.